app/models/program.rb

```diff
@@ -199,4 +204,30 @@ def check_languages_format
     # We check if every language is a valid ISO 639-1 language
     errors.add(:languages, 'must be ISO 639-1 valid codes') unless languages_array.select{ |x| ISO_639.find(x).nil? }.empty?
   end
+
+  ##
+  # Check if schedule_interval is a divisor of 60 minutes
+  #
+  def schedule_interval_divisor_60
+    errors.add(:schedule_interval, 'must be a divisor of 60') if schedule_interval > 0 && 60 % schedule_interval > 0
+  end
+
+  ##
+  # Unschedule all the events which don't fit
+  #
+  def unschedule_unfit_events
```

Comment:
I destroy all event schedules with start time is not suitable for new schedule interval. For example: 
schedule interval was 15 minutes
event1 start time = 9:15
event2 start_time = 9:30
new schedule interval = 10 minutes
Only event1 will be destroyed because 9:30 can be start time with new schedule interval 10 minutes